cdisc_rules_engine/services/data_readers/dataset_json_reader.py

```diff
@@ -0,0 +1,58 @@
+import pandas as pd
+import json
+
+from cdisc_rules_engine.interfaces import (
+    DataReaderInterface,
+)
+
+# Testing only
+# test_file_path = "D:\\CDISC_CORE_Engine\\Python_work_Jozef\\lb.json"
+
+
+class DatasetJSONReader(DataReaderInterface):
+    def from_file(self, file_path):
+        # print("dataset_json_reader: starting reading_from_file = ", file_path)
+        f = open(file_path)
+        # returns JSON object as
+        # a dictionary
+        data = json.load(f)
+        # print("data = ", data)
+        # check for "clinicalData"
+        start_data_string = "clinicalData"
+        if 'clinicalData' in data:
+            start_data_string = "clinicalData"
+        else:
+            # print("could not find clinicalData key in file - trying referenceData")
+            start_data_string = "referenceData"
+        # we need to know the OID of the "ItemGroupData"
+        # TODO: very probably, this can be much simpler ...
+        item_group_data = data[start_data_string]["itemGroupData"]
+        item_group_oids = item_group_data.keys()
```

Comment:
What are the other possible keys?

Reply:
If there is just one "domain" represened by the dataset, there is only one key. But people may add data from other domains in the same dataset, which would cause several keys. This can/should not be covered by the engine.
So to be sure to not cause an exception, we take the first key, which is the OID of the ItemGroup for which the definition is in the define.xml.